packages/ember-data/lib/serializers/rest_serializer.js

```diff
@@ -487,20 +490,22 @@ var RESTSerializer = JSONSerializer.extend({
     var payload = this.normalizePayload(rawPayload);
 
     for (var prop in payload) {
-      var typeName = this.typeForRoot(prop);
-      if (!store.modelFactoryFor(typeName, prop)) {
-        Ember.warn(this.warnMessageNoModelForKey(prop, typeName), false);
-        continue;
-      }
-      var type = store.modelFor(typeName);
-      var typeSerializer = store.serializerFor(type);
+      if (prop !== "meta") {
```

Comment:
Instead of doing this check could you call this.extractMeta before this.normalizePayload to mirror how meta gets extracted in the normal flow?